packages/orchestrator/lib/client.integration.test.ts

```diff
@@ -31,94 +31,102 @@ describe('OrchestratorClient', async () => {
+                process: async (task) => {
+                    await scheduler.succeed({ taskId: task.id, output });
                 }
             });
-            expect(res.unwrap()).toEqual(output);
-        } finally {
-            processor.stop();
-        }
-    });
-    it('should return an error if execute fails', async () => {
-        const groupKey = 'groupC';
-
-        const processor = new MockProcessor({
-            groupKey,
-            process: async (task) => {
-                await scheduler.fail({ taskId: task.id });
+            try {
+                const res = await client.execute({
+                    name: 'Task',
+                    groupKey: groupKey,
+                    args: {
+                        name: 'Action',
+                        connection: {
+                            id: 1234,
+                            provider_config_key: 'P',
+                            environment_id: 5678
+                        },
+                        activityLogId: 9876,
+                        input: { foo: 'bar' }
+                    }
+                });
+                expect(res.unwrap()).toEqual(output);
+            } finally {
+                processor.stop();
             }
         });
-        try {
-            const res = await client.execute({
-                name: 'Task',
-                groupKey: groupKey,
-                args: {
-                    name: 'Action',
-                    connection: {
-                        id: 1234,
-                        provider_config_key: 'P',
-                        environment_id: 5678
-                    },
-                    activityLogId: 9876,
-                    input: { foo: 'bar' }
+        it('should return an error if task fails', async () => {
+            const groupKey = 'groupC';
+
+            const errorPayload = { message: 'something bad happened' };
+            const processor = new MockProcessor({
+                groupKey,
+                process: async (task) => {
+                    await scheduler.fail({ taskId: task.id, error: errorPayload });
                 }
             });
-            expect(res.isErr()).toBe(true);
-        } finally {
-            processor.stop();
-        }
+            try {
+                const res = await client.execute({
+                    name: 'Task',
+                    groupKey: groupKey,
+                    args: {
+                        name: 'Action',
+                        connection: {
+                            id: 1234,
+                            provider_config_key: 'P',
+                            environment_id: 5678
+                        },
+                        activityLogId: 9876,
+                        input: { foo: 'bar' }
+                    }
+                });
+                expect(res.isOk()).toBe(false);
+                if (res.isErr()) {
+                    expect(res.error.payload).toBe(res.error.payload);
```

Comment:
this `if` is actually the only addition in the file. The rest is just grouping tests